File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let sentinel=new ListNode();
    sentinel.next=head;
    let f=sentinel;
    for(let i=0;i<n;i++){
        f=f.next;
    }

    let s=sentinel;
    while(f.next){
        s=s.next;
        f=f.next;
    }
    s.next=s.next.next;
    return sentinel.next;

    
};

File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode sentinel=new ListNode(0);
        sentinel.next=head;
        ListNode curr=head;
        int length=0;
        while(curr!=null){
            length++;
            curr=curr.next;
        }
        ListNode prev=sentinel;
        for(int i=0;i<length-n;i++){
            prev=prev.next;
        }
        prev.next=prev.next.next;
        return sentinel.next;

    }
}